lib/backports/2.7.0/enumerable/filter_map.rb

```diff
@@ -0,0 +1,12 @@
+unless Enumerable.method_defined? :filter_map
+  module Enumerable
+    def filter_map
```

Comment:
My gut feeling is that doing `map(&block).compact` would be faster.

Haven't checked, but are the specs checking for enumerables yielding more than 1 element? The treatment of those is often problematic.

Reply:
> My gut feeling is that doing `map(&block).compact` would be faster.

It would be wrong, as `filter_map` also drops `false`s:
```ruby
%w[1 nil false Time.now].filter_map { eval(_1) }
# => [1, 2020-02-03 13:13:38.597956549 +0200] 
%w[1 nil false Time.now].map { eval(_1) }.compact
# => [1, false, 2020-02-03 13:13:48.794758541 +0200] 
```

> Haven't checked, but are the specs checking for enumerables yielding more than 1 element?

Nope. But it doesn't seem to produce any problems that I can think of... Like, 
```ruby
(1..3).each_with_object('x').filter_map { |i, x| x if i.odd? }
# => ["x", "x"] -- both in 2.7 and backports.
```

Reply:
Indeed, my bad.